### PR TITLE
bats: Factor setup and teardown logic out of tests, greatly reducing duplication

### PR DESCRIPTION
**Pull request by mjesuele**
This has been bothering me for a while.

### CODE REVIEW

File: bats/helper/common.bash [hunk 1]
```diff
@@ -5,10 +5,33 @@ if [ -z "$BATS_TMPDIR" ]; then
...
+	export PATH=$PATH:~/go/bin
+	export NOMS_VERSION_NEXT=1
+	cd $BATS_TMPDIR
+	# Append the directory name with the pid of the calling process so
+	# multiple tests can be run in parallel on the same machine
+	mkdir "dolt-repo-$$"
+	cd "dolt-repo-$$"
+
+	SKIP_INIT=$1
+	if [ "$SKIP_INIT" != true ] ; then
```
Comment (mjesuele) [inline bats/helper/common.bash]: Please do let me know if you know a better way to write this. I tried a few, actually...

Comment (timsehn) [reply bats/helper/common.bash]: What if you made two functions? One called setup_common() and one called setup_no_dolt() and make the setup_no_dolt() call setup_common(). Passing the variable is a little janky. The above seems to be more readable.

File: bats/helper/common.bash [hunk 2]
```diff
@@ -5,10 +5,33 @@ if [ -z "$BATS_TMPDIR" ]; then
...
 batshelper() { echo `nativebatsdir helper/$1`; }
+
+set_dolt_user() {
+	dolt config --global --add user.name "$1" > /dev/null 2>&1
+	dolt config --global --add user.email "$2" > /dev/null 2>&1
+}
+
+setup_common() {
+	export PATH=$PATH:~/go/bin
+	export NOMS_VERSION_NEXT=1
```
Comment (timsehn) [inline bats/helper/common.bash]: I think this NOMS_VERSION_NEXT is not be needed anymore.

File: bats/helper/common.bash (hunk 1 above)
Comment (mjesuele) [reply bats/helper/common.bash]: I like that, thanks.

**Pull request merged**